app/controllers/api/v2/override_values_controller.rb

```diff
@@ -11,6 +12,8 @@ class OverrideValuesController < V2::BaseController
       before_action :return_if_smart_mismatch, :only => [:index, :create, :show, :update, :destroy]
       before_action :return_if_override_mismatch, :only => [:show, :update, :destroy]
 
+      deprecate_param [:create, :update], [:override_value, :use_puppet_default], :skip_foreman
```

Comment:
I'm against introducing a 60 line, very complex module, for something that should not occur more then a handful of times at most and can be done in 5 easy to read and very explicit lines, even if they repeat 5 times in the code base. Readability in my opinion is more important here then then DRYing the code.